helios-core/src/main/kotlin/helios/instances/instances.kt

```diff
@@ -186,4 +187,44 @@ interface TripleDecoderInstance<out A, out B, out C> : Decoder<Triple<A, B, C>>
       }
   }
 
+}
+
+@extension
+interface NullableEncoderInstance<in A> : Encoder<A?> {
+
+  fun encoderA(): Encoder<A>
+
+  override fun A?.encode(): Json {
+    val maybeThis = toOption()
+    return Option.encoder(encoderA()).run { maybeThis.encode() }
```

Comment:
```suggestion
    return this?.let { a ->encoderA().let { a.encode() } } ?: JsNull
```

Need multi-line suggestions ><

There is not need for this indirection, it's redundant overhead and when parsing large `Json`s will unnecessary slow down perf.